### DISCUSSION TITLE
customize how builder (m2a) displays a wysiwyg-field

### DISCUSSION THREAD

**Discussion by kamil-malinski** [directus/directus#12384] [Archive]
Hi!

Is there a way to change how the builder (m2a) displays a WYSIWYG-field? I know that you can change between displaying "raw value" and "related value". "raw value" will simply output the HTML-text (with encoded html entities) and "related value" will throw out all of the formatting and display everything in one line:

![CleanShot 2022-03-25 at 14 04 01](https://user-images.githubusercontent.com/68733431/160125993-32daf8be-6f75-4b17-83bd-58c25d29fd2f.png)

What I would like to show here is the fully formatted text as it was entered and formatted in the WYSIWYG-editor and not just a single line without all the formatting.

Do I need a custom interface for that?

Cheers

MJFox

**Comment by rijkvanzanten** [comment]
> What I would like to show here is the fully formatted text as it was entered and formatted

Do you mean you'd like the whole thing to show up in multiple lines? The `formatted-value` display is probably the closest you can get right now, but that will still render it as a single line preview 🤔

**Reply by kamil-malinski** [reply]
> Do you mean you'd like the whole thing to show up in multiple lines?

Yes. I'm using the builder (m2a) to create dynamic content for an article body. The editor is able to create the article body from various components ("headline", "paragraph", "quote" etc.). I'm using the WYSIWYG-field for the paragraph-component.

The idea is that the editor doesn't need to open each component to see what's inside and sees the article body as a whole.

Cheers

MJFox

**Reply by rijkvanzanten** [reply]
Gotcha! Displays are currently single-line only, so there's no way to have the full article body previewed in that row without creating a custom interface to do so 👍🏻

**Reply by kamil-malinski** [reply]
> Gotcha! Displays are currently single-line only, so there's no way to have the full article body previewed in that row without creating a custom interface to do so 👍🏻

ok, thanks for confirming that

since I'm new to directus: could you please give me a nod in the right direction? Do I need to create a custom interface which is basically the same as the default wysiwyg-interface but with a custom display that displays the whole field-value without any filtering? would that work like that?

**Reply by rijkvanzanten** [reply]
The many to any interface renders the preview of the item using _Displays_. A _Display_ is the "inline representation" of the saved value (so a little preview for an image, or the single-line text without html characters for a wysiwyg). You should be able [to create your own Display](https://docs.directus.io/extensions/displays/) and use that in the configuration for your table 🙂

**Reply by kamil-malinski** [reply]
thanks, that gives me an idea about the concept behind it

> and use that in the configuration for your table

this part I don't get... what is the "configuration of my table"? is this the "display_template"-field in the "collections"-table?

**Reply by rijkvanzanten** [reply]
Yeah! That's exactly it 🙂 By configuring your wywisywg field to use your custom display, it will be used wherever the display template is used, including in the m2a row items 👍🏻

**Reply by kamil-malinski** [reply]
awesome, thanks a lot! 👍🏻

**Reply by kamil-malinski** [reply]
just tried what you suggested and it works great (together with a few css-overrides):

```vue
<template>
  <div v-html="value"></div>
</template>

<script>
export default {
  props: {
    value: {
      type: String,
      default: null,
    },
  },
};
</script>
```

thanks again!

**Reply by marcusand** [reply]
hey @kamil-malinski would you mind also sharing your css-overrides? trying the same thing here :)

**Reply by chris-saunders-rocketlab** [reply]
> Yeah! That's exactly it 🙂 By configuring your wywisywg field to use your custom display, it will be used wherever the display template is used, including in the m2a row items 👍🏻

How exactly do you configure your field to use the custom display?
"the "display_template"-field in the "collections"-table" contains the template string (like: {{name}}). 
This does not look like a way to specify the display to use.

The directus_fields table has a column called 'display' which does look like a way to specify the display to use for the field in question however, changing this value has so far only made the Display tab of the field blank and unusable while the M2A field still uses the display that is no longer specified.

Also, while Directus loads the display extension it is not available in the display configuration for the field. No idea if it's supposed to.

**Reply by chris-saunders-rocketlab** [reply]
Well I got the display to be available in Directus but when I choose it, the M2A lists' behaviour does not change at all, it displays just as if I still had Related Value selected and definitely does not show the template from the display extension I've created.

**Reply by chris-saunders-rocketlab** [reply]
> Yeah! That's exactly it 🙂 By configuring your wywisywg field to use your custom display, it will be used wherever the display template is used, including in the m2a row items 👍🏻

Does anyone know how to do this?

None of the above seems to make sense.

I have an M2A field which supports 3 collections. Let's just call them A, B & C.
Right now in the M2A field, each item is listed as COLLECTION: DISPLAY_TEMPLATE_FOR_COLLECTION.
Since the display template of collection A is Name they look like A: Fred, B: Sally etc.

I now need to add an indicator to warn users that item A: Fred is used in multiple locations so by editing this item, all locations will be updated.

I have created a custom display for the M2A field however nothing changes. 
It's vue template is definitely not used.

Is there any documentation anywhere that details not just what to do but how?

**Reply by rijkvanzanten** [reply]
> I now need to add an indicator to warn users that item A: Fred is used in multiple locations so by editing this item, all locations will be updated.

I'd recommend putting a Notice interface at the top of the form in that case, easier than trying to hack around with an unused field + custom display 👍🏻

**Reply by chris-saunders-rocketlab** [reply]
Right yes I thought of something along those lines since my last post.

It is preferable for us to be able to show the indicator in the M2A list, before the user goes to edit the item but something at the top of the form would suffice (and probably both would be best).

We've used the Button interface to create a link to preview the page that utilises the collection item in question, so my next line of enquiry was going to be a custom interface along those lines.

I'm not sure I can use the built in Notice interface since I have to perform a query against the API to know if that item is used elsewhere (IE was added as existing not created as new). The Notice interface appears to be text entry only, but I can see how a custom interface might sort out my issue as a field in the form.
